dds/DCPS/FilterEvaluator.cpp

```diff
@@ -125,11 +125,22 @@ FilterEvaluator::SerializedForEval::lookup(const char* field) const
     return iter->second;
   }
   Message_Block_Ptr mb (serialized_->duplicate());
-  // TODO(iguessthislldo): Use Encap
   Serializer ser(mb.get(),
     cdr_ ? Encoding::KIND_XCDR1 : Encoding::KIND_UNALIGNED_CDR, swap_);
   if (cdr_) {
-    ser.skip(4); // CDR encapsulation header
+    EncapsulationHeader encap;
+    if (!(ser >> encap)) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR ")
+        ACE_TEXT("%CDataReaderImpl::lookup_instance: ")
+        ACE_TEXT("deserialization of encapsulation header failed.\n")));
+      //  TraitsType::type_name()));
```

Comment:
You don't have to have it say the name, that's just the convention in the DataReader/WriterImpl_Ts, just `FilterEvaluator::SerializedForEval::lookup` should be fine.